sdk/resources/arm-resources/src/resourceManagementClientContext.ts

```diff
@@ -1,74 +1,62 @@
+   * @param credentials Subscription credentials which uniquely identify client subscription.
+   * @param subscriptionId The Microsoft Azure subscription ID.
+   * @param options The parameter options
    */
   constructor(
-    credentials: msRest.ServiceClientCredentials | TokenCredential,
+    credentials: coreAuth.TokenCredential,
     subscriptionId: string,
-    options?: Models.ResourceManagementClientOptions
+    options?: ResourceManagementClientOptionalParams
   ) {
-    if (credentials == undefined) {
-      throw new Error("'credentials' cannot be null.");
+    if (credentials === undefined) {
+      throw new Error("'credentials' cannot be null");
     }
-    if (subscriptionId == undefined) {
-      throw new Error("'subscriptionId' cannot be null.");
+    if (subscriptionId === undefined) {
+      throw new Error("'subscriptionId' cannot be null");
     }
 
+    // Initializing default values for options
     if (!options) {
       options = {};
     }
-    if (!options.userAgent) {
-      const defaultUserAgent = msRestAzure.getDefaultUserAgentValue();
-      options.userAgent = `${packageName}/${packageVersion} ${defaultUserAgent}`;
+    const defaults: ResourceManagementClientOptionalParams = {
+      requestContentType: "application/json; charset=utf-8",
+      credential: credentials
+    };
+
+    const packageDetails = `azsdk-js-arm-resources/1.0.0-beta.1`;
+    const userAgentPrefix =
+      options.userAgentOptions && options.userAgentOptions.userAgentPrefix
+        ? `${options.userAgentOptions.userAgentPrefix} ${packageDetails}`
+        : `${packageDetails}`;
+
+    if (!options.credentialScopes) {
+      options.credentialScopes = ["https://management.azure.com/.default"];
     }
-
-    super(credentials, options);
-
-    this.apiVersion = "2021-01-01";
-    this.acceptLanguage = "en-US";
-    this.longRunningOperationRetryTimeout = 30;
-    this.baseUri = options.baseUri || this.baseUri || "https://management.azure.com";
-    this.requestContentType = "application/json; charset=utf-8";
-    this.credentials = credentials;
+    const optionsWithDefaults = {
+      ...defaults,
+      ...options,
+      userAgentOptions: {
+        userAgentPrefix
+      },
+      baseUri: options.endpoint || "https://management.azure.com"
+    };
+    super(optionsWithDefaults);
+    // Parameter assignments
     this.subscriptionId = subscriptionId;
 
-    if (options.acceptLanguage !== null && options.acceptLanguage !== undefined) {
-      this.acceptLanguage = options.acceptLanguage;
-    }
-    if (
-      options.longRunningOperationRetryTimeout !== null &&
-      options.longRunningOperationRetryTimeout !== undefined
-    ) {
-      this.longRunningOperationRetryTimeout = options.longRunningOperationRetryTimeout;
-    }
+    // Assigning values to Constant parameters
+    this.$host = options.$host || "https://management.azure.com";
+    this.apiVersion = options.apiVersion || "30.0.0-beta.1";
```

Comment:
are we sure about using `30.0.0-beta.1` here?

Reply:
The generated one by codegen is `this.apiVersion = options.apiVersion || "1.0.0-beta.1";`. I think here `options.apiVersion` shouldn't be `undefined`, so now I don't understand the meaning of version in the end. could you please give a detailed explanation. Thanks